app/src/main/java/com/android/streetworkapp/model/event/EventRepositoryFirestore.kt

```diff
@@ -156,6 +156,42 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
     }
   }
 
+  /**
+   * Remove a participant from all events and delete events where the user is the owner.
+   *
+   * @param uid The user ID.
+   */
+  override suspend fun removeParticipantFromAllEvents(uid: String) {
+    require(uid.isNotEmpty())
+    try {
+      val events =
+          db.collection(COLLECTION_PATH)
+              .whereArrayContains(FIELD_LIST_PARTICIPANTS, uid)
```

Comment:
Suggestion: this is more of an edge case but it might problematic if a uid contains the same letters as another. To prevent that, you can use `whereEqualTo` instead

Reply:
The `whereArrayContainsAny()` function definition is the following:
> Creates and returns a new Query with the additional filter that documents must contain the specified field, the value must be an array, and that the array must contain the provided value.

This function check for arrays (in our case the participants list) that contains a certain value (in our case the participant to remove) and the check is made on the entire value according to its documentation. In this case I think that my implementation is correct, what do you think of it?

Reply:
You are right, i misread what the function was actually doing, my apologies